File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    conversation = Conversation.find(params[:conversation_id])
    stream_for conversation
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def send_message(data)
    conversation = Conversation.find(data["conversation_id"])
    message = conversation.messages.create!(
      user_id: current_user.id,
      content: data["content"]
    )
    ChatChannel.broadcast_to(conversation, message: message.as_json)
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

const conversationId = 1

consumer.subscriptions.create({channel: "ChatChannel", conversation_id: conversationId}, {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log("New message:", data)
    // Called when there's incoming data on the websocket for this channel
  },

  send_message(content) {
    this.perform("send_message", { conversation_id: conversationId, content: content })
  }
});
